src/eos/eosEngine.js

```diff
@@ -532,86 +532,107 @@ export class EosEngine extends CurrencyEngine {
-                  this.log.warn(
-                    `Account created with accountName: ${accountName} and transactionId: ${transactionId}`
-                  )
-                }
-                throw new Error(
-                  `${server} could not find account with public key: ${params[0]}`
+                  }
+                )
+                const response = await request.json()
+                const { accountName, transactionId } = response
+                if (!accountName) throw new Error(response)
+                this.log.warn(
+                  `Account created with accountName: ${accountName} and transactionId: ${transactionId}`
                 )
               }
-              const accountName = authorizersData.account_names[0]
-              const getAccountBody = JSON.stringify({
-                account_name: accountName
-              })
-              const accountReply = await this.eosJsConfig.fetch(
-                `${server}/v1/chain/get_account`,
-                {
-                  method: 'POST',
-                  body: getAccountBody
-                }
+              throw new Error(
+                `${server} could not find account with public key: ${params[0]}`
               )
-              if (!accountReply.ok) {
-                throw new Error(
-                  `${server} get_account failed with ${authorizersReply}`
-                )
+            }
+            const accountName = authorizersData.account_names[0]
+            const getAccountBody = JSON.stringify({
+              account_name: accountName
+            })
+            const accountReply = await this.eosJsConfig.fetch(
+              `${server}/v1/chain/get_account`,
+              {
+                method: 'POST',
+                body: getAccountBody
               }
-              return { server, result: await accountReply.json() }
+            )
+            if (!accountReply.ok) {
+              throw new Error(
+                `${server} get_account failed with ${authorizersReply}`
+              )
             }
-          )
+            return { server, result: await accountReply.json() }
+          }
+        )
+        // dfuse API is EOS only
+        const dfuseFuncs = this.currencyInfo.defaultSettings.otherSettings.eosDfuseServers.map(
+          server => async () => {
+            if (this.currencyInfo.currencyCode !== 'EOS')
+              throw new Error('dfuse only supports EOS')
+            const response = await this.eosJsConfig.fetch(
+              `${server}/v0/state/key_accounts?public_key=${params[0]}`
+            )
+            if (!response.ok) {
+              throw new Error(
+                `${server} get_account failed with ${response.code}`
+              )
+            }
+            const responseJson = await response.json()
+            return {
+              server,
+              result: {
+                account_name: responseJson.account_names[0]
```

Comment:
Might be worth checking that the account name is not undefined since there may be no elements in the array